src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs

```diff
@@ -264,5 +300,13 @@ private void ThrowForModifiedManagedSslOptionsIfStarted()
             // SslOptions is changed, since SslOptions itself does not do any such checks.
             _underlyingHandler.SslOptions = _underlyingHandler.SslOptions;
         }
+
+        private void CheckDisposed()
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException(GetType().ToString());
```

Comment:
Wouldn't be calling ```GetType().ToString()``` expensive, why not use ```typeof```` directly

Reply:
GetType() will be the derived type if there is one. And while it's more expensive than a const, it pales in comparison to the cost of the exception that's about to be thrown.